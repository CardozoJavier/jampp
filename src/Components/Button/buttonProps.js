import {
  AccessoryButtonContainer,
  SecondaryButtonContainer,
  PrimaryButtonContainer,
  OptionButtonContainer,
  StatusButtonContainer,
  LinkButtonContainer,
} from './styles';


const buttonProps = {
  "accessory-default-left": {
    defaultClassName: 'button accessory--default-left',
    optionalClassName: '',
    iconClassName: 'icon accessory--default-left',
    ButtonContainer: AccessoryButtonContainer,
    iconProps: {
      // left: '10px',
      // width: '11px',
      // height: '11px',
    },
  },
  "accessory-gray-left": {
    defaultClassName: 'button accessory--gray-left',
    optionalClassName: '',
    iconClassName: 'icon accessory--gray-left',
    ButtonContainer: AccessoryButtonContainer,
    iconProps: {
      // left: '10px',
      // width: '11px',
      // height: '11px',
    },
  },
  "accessory-default-both": {
    defaultClassName: 'button accessory--default-both',
    optionalClassName: '',
    ButtonContainer: AccessoryButtonContainer,
  },
  "accessory-inverted-left": {
    defaultClassName: 'button accessory--inverted-left',
    optionalClassName: '',
    ButtonContainer: AccessoryButtonContainer,
  },
  "secondary-default-medium": {
    defaultClassName: 'button secondary--default-medium',
    optionalClassName: '',
    ButtonContainer: SecondaryButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "secondary-gray-medium": {
    defaultClassName: 'button secondary--gray-medium',
    optionalClassName: '',
    ButtonContainer: SecondaryButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "secondary-inverted-medium": {
    defaultClassName: 'button secondary--inverted-medium',
    optionalClassName: '',
    ButtonContainer: SecondaryButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "primary-default-large": {
    defaultClassName: 'button primary--default-large',
    optionalClassName: '',
    ButtonContainer: PrimaryButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "primary-inverted-large": {
    defaultClassName: 'button primary--inverted-large',
    optionalClassName: 'button primary--inverted-large__selected',
    ButtonContainer: PrimaryButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "label-default-small": {
    defaultClassName: 'button label--default-small',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: 'icon label--default-small__right',
    iconProps: {
      // right: '8px',
      // width: '4px',
      // height: '4px',
    },
  },
  "label-default-medium": {
    defaultClassName: 'button label--default-medium right-icon',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: 'icon label--default-medium__right',
    iconProps: {
      // right: '8px',
      // width: '6px',
      // height: '6px',
    },
  },
  "link-default-left": {
    defaultClassName: 'button link--default-left',
    optionalClassName: '',
    ButtonContainer: LinkButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "status-small-green": {
    defaultClassName: 'button status--small-green',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "status-small-red": {
    defaultClassName: 'button status--small-red',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "status-small-yellow": {
    defaultClassName: 'button status--small-yellow',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "status-small-blue": {
    defaultClassName: 'button status--small-blue',
    optionalClassName: '',
    ButtonContainer: StatusButtonContainer,
    iconClassName: '',
    iconProps: {},
  },
  "option-default-medium": {
    defaultClassName: 'button option--default-medium',
    optionalClassName: 'button option--default-medium__selected',
    ButtonContainer: OptionButtonContainer,
    iconClassName: '',
    iconProps: {},
  },

};

export default buttonProps;
