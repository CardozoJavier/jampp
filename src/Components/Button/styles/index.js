export { default as ButtonInput } from './ButtonInput.styles';
export { default as LinkButtonContainer } from './LinkButtonContainer.styles';
export { default as OptionButtonContainer } from './OptionButtonContainer.styles';
export { default as PrimaryButtonContainer } from './PrimaryButtonContainer.styles';
export { default as SecondaryButtonContainer } from './SecondaryButtonContainer.styles';
export { default as AccessoryButtonContainer } from './AccessoryButtonContainer.styles';
export { default as SecondaryActionButtonContainer } from './SecondaryActionButtonContainer.styles';
