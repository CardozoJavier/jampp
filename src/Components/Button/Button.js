import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { ButtonInput } from './styles';
import buttonProps from './buttonProps';
import { IconGenerator } from '../UI/Icons';
import { bemDestruct, getClassName } from '../../utils';

/**
 * Button component should be called with
 * @param {String} type - (Required) Define button classes for styling.
 * @param {String} label - (Optional) Text to be display inside button.
 * @param {Node} children - (Optional) HTML element to be rendered, e.g.: strong tag inside button.
 * @param {Function} icon - (Optional) Function that returns an svg icon.
 * @param {Boolean} disabled - (Optional) Disable component if is true.
 * @param {Function} onClick - (Optional) Callback to trigger on onClick event.
 * @param {Function} onClickIcon - (Optional) Callback to trigger on icon onClick event.
 * @param {Boolean} fixed - (Optional) Disabled move up transform on hover.
 * @return {React Component} A view for button.
 */
const Button = ({ type, label, children, icon, disabled, onClick, onClickIcon, fixed }) => {
  const { defaultClassName, optionalClassName, ButtonContainer, iconProps, iconClassName } = buttonProps[type];

  const [className, setClassName] = useState(defaultClassName);
  const toggleToClassName = getClassName(className, defaultClassName, optionalClassName);

  const handleClick = () => {
    onClick && onClick();
    setClassName(toggleToClassName);
  };

  return (
    <ButtonContainer
      onClick={disabled ? null : handleClick}
      className={bemDestruct(className, disabled)}
      disabled={disabled}
      fixed={fixed}
    >
      <ButtonInput children={label || children} />
      {icon &&
        <IconGenerator
          renderIcon={icon}
          props={{
            ...iconProps,
            className: bemDestruct(iconClassName, disabled),
            onClick: onClickIcon
          }}
        />
      }
    </ButtonContainer>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string,
  children: PropTypes.node,
  icon: PropTypes.func,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  onClickIcon: PropTypes.func,
  fixed: PropTypes.bool,
};

Button.defaultProps = {
  label: '',
  children: null,
  disabled: false,
  onClick: () => null,
  onClickIcon: () => null,
  fixed: false,
};

export default Button;
