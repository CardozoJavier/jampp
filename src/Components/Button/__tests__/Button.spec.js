/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Button from '../Button';

describe('<Button/> component', () => {
  const type = 'primary-default-large';
  const children = 'Button text';

  test('should display the text', () => {
    render(<Button type={type}>{children}</Button>);
    expect(screen.queryByText(children)).not.toBeNull();
  });

  test('should call onClick callback', () => {
    const handleClick = jest.fn();
    render(<Button type={type} onClick={handleClick}>{children}</Button>);
    fireEvent.click(screen.getByText(children));
    expect(handleClick).toBeCalled();
  });

  test('should match snapshot', () => {
    const { container } = render(<Button type={type}>{children}</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
