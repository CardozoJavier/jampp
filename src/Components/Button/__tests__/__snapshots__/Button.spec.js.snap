// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component should match snapshot 1`] = `
<div
  class="sc-jAaTju klJtKp 
    button
    primary
    button-large
    primary--default
    primary--default__undefined
    
    button primary--default-large
  "
>
  <a
    class="sc-bdVaJa gYYOnH"
  >
    Button text
  </a>
</div>
`;
